be/src/exprs/math_functions.cpp

```diff
@@ -746,4 +747,101 @@ StatusOr<ColumnPtr> MathFunctions::rand_seed(FunctionContext* context, const Col
     return rand(context, columns);
 }
 
+template <LogicalType TYPE, bool isNorm>
```

Comment:
1. What's the usage of this function?
2. How large two vectors it can support?

Reply:
The function is to compute the cosine similarity of two vectors. I've described usage in a PR comment.

It can support the arbitrary length of two vectors.